src/stats.ts

```diff
@@ -79,43 +80,50 @@ export class Stats {
     return this._openPositionCount;
   }
 
-  /**
-   * Fetches all stats.
-   */
-  public async fetch() {
-    await Promise.all([
-      this.fetchCollateralSupply(),
-      this.fetchDebtSupply(),
-      this.fetchBorrowingRate(),
-      this.fetchOpenPositionCount(),
-    ]);
-  }
-
   /**
    * Fetches current collateral supply (Amount of wstETH locked in Raft protocol).
+   * @returns Fetched collateral supply
    */
-  private async fetchCollateralSupply() {
+  async fetchCollateralSupply(): Promise<Decimal> {
     this._collateralSupply = new Decimal(await this.raftCollateralToken.totalSupply(), Decimal.PRECISION);
+
+    return this._collateralSupply;
   }
 
   /**
    * Fetches current debt supply (Amount of R users borrowed).
+   * @returns Fetched debt supply
    */
-  private async fetchDebtSupply() {
+  async fetchDebtSupply(): Promise<Decimal> {
     this._debtSupply = new Decimal(await this.raftDebtToken.totalSupply(), Decimal.PRECISION);
+
+    return this._debtSupply;
   }
 
   /**
-   * Fetches current borrowing rate.
+   * Fetches current borrowing rate for specified collateral token.
+   * @param collateralToken Collateral token to fetch borrowing rate for.
+   * @returns Fetched borrowing rate.
    */
-  private async fetchBorrowingRate() {
-    this._borrowingRate = new Decimal(await this.positionManager.getBorrowingRate(), Decimal.PRECISION);
+  async fetchBorrowingRate(collateralToken: CollateralToken): Promise<Decimal> {
```

Comment:
Only wstETH can be used here

```suggestion
  async fetchBorrowingRate(collateralToken: UnderlyingCollateralToken): Promise<Decimal> {
```